linq4j/src/test/java/org/apache/calcite/linq4j/test/ExpressionTest.java

```diff
@@ -1719,6 +1724,60 @@ public void checkBlockBuilder(boolean optimizing, String expected) {
         Expressions.toString(Expressions.constant(set)));
   }
 
+  @Test void testLargeMethod() {
+//     public class MyClass {
+//       public void hugeMethod() {
+//         if (true) {
+//           // Long variable name declaration 1;
+//         } else if (false) {
+//           // Long variable name declaration 2;
+//         } else if (false) {
+//           // Long variable name declaration 3;
+//         } else if (false) {
+//           // Long variable name declaration 4;
+//         } else
+//           // Long variable name declaration 4;
+//         }
+//       }
+
+    // Generate long variable names.
+    final String longName = StringUtils.repeat('a', 5000);
+
+    DeclarationStatement longDecl =
+        Expressions.declare(0, longName, Expressions.constant(10));
+    final ConditionalStatement ifThenElse =
+        Expressions.ifThenElse(
+            Expressions.constant(true),
+            longDecl,
+            Expressions.constant(false),
+            longDecl,
+            Expressions.constant(false),
+            longDecl,
+            longDecl);
+
+    final MethodDeclaration hugeMethod =
+        Expressions.methodDecl(Modifier.PUBLIC,
+        Void.TYPE,
+        "hugeMethod",
+        Collections.emptyList(),
+        Blocks.toFunctionBlock(ifThenElse));
+
+    final ClassDeclaration classDecl =
+        new ClassDeclaration(Modifier.PUBLIC,
+        "MyClass",
+        null,
+        ImmutableList.of(),
+        ImmutableList.of(hugeMethod));
+
+    final String originalClass = Expressions.toString(classDecl);
+    final String classWithMethodSplitting = Expressions.toString(classDecl, 4000);
+
+    // Some local variables in hugeMethod have been extracted to be fields named local$<number> in
+    // the enclosing class definition.
+    Assertions.assertFalse(originalClass.contains("local$"));
```

Comment:
Use `assertThat` rather than `assertFalse` and `assertTrue`.

Reply:
Done